### PR TITLE
Implementation of Mergeable and SetOperations for pilosa's lookups

### PR DESCRIPTION
**Pull request by kuba--**
! Should be merged after: https://github.com/src-d/go-mysql-server/pull/277 !

This PR. Implements 2 interfaces (Mergeable and SetOperations) for following lookups:
- indexLookup
- ascendLookup
- descendLookup

It closes: https://github.com/src-d/go-mysql-server/issues/212

### simplified example

```      
       unionLookup := lookup0.Union(lookup1, lookup2)

       ascLookup := pilosaIdx.AscendLessThan(keys...) 
       diffLookup := ascLookup.Difference(unionLookup)
```

### CODE REVIEW

**Comment by kuba--** [comment]
Ok, right now is  squashed and rebased (+ some fixes)

File: engine.go [hunk 1]
```diff
@@ -16,19 +16,18 @@ type Engine struct {
 }
 
 // New creates a new Engine
-func New(c *sql.Catalog, a *analyzer.Analyzer, versionPostfix string) *Engine {
+func New(c *sql.Catalog, a *analyzer.Analyzer) *Engine {
```
Comment (ajnavarro) [inline engine.go]: that code is not well rebased, you are removing changes from master.

**Comment by kuba--** [comment]
Ok, now it looks it contains only my chnages.

File: sql/index/pilosa/driver_test.go [hunk 2]
```diff
@@ -349,6 +350,364 @@ func TestAscendDescendIndex(t *testing.T) {
...
+
+	interLookup, ok = lookupPath.(sql.SetOperations)
+	require.True(ok)
+	interIt, err = interLookup.Intersection(lookupLang).Values()
+	require.NoError(err)
+	loc, err := interIt.Next()
+	require.NoError(err)
+	require.True(bytes.Equal(loc, itPath.records[0].location))
+	_, err = interIt.Next()
+	require.True(err == io.EOF)
```
Comment (ajnavarro) [inline sql/index/pilosa/driver_test.go]: maybe you can use require.Equal in that cases.

Comment (kuba--) [reply sql/index/pilosa/driver_test.go]: I afraid that it won't work for byte slices, but looks that it works, so I've updated.

**Pull request merged**